File: CRM/backend/scripts/smoke_dashboard.py
"""Smoke test for dashboard visits endpoints."""

from __future__ import annotations

import os
import sys

import httpx

API_BASE_URL = os.getenv("API_BASE_URL", "http://127.0.0.1:8000/api/v1").rstrip("/")
EMAIL = os.getenv("SMOKE_LOGIN_EMAIL", "admin@example.com")
PASSWORD = os.getenv("SMOKE_LOGIN_PASSWORD", "password")
TIMEOUT = float(os.getenv("SMOKE_LOGIN_TIMEOUT", "5"))


def login() -> str | None:
    url = f"{API_BASE_URL}/auth/login"
    try:
        response = httpx.post(url, json={"email": EMAIL, "password": PASSWORD}, timeout=TIMEOUT)
    except Exception as exc:  # noqa: BLE001
        print(f"[smoke] Login request failed: {exc}")
        return None

    if response.status_code != 200:
        print(f"[smoke] Login failed ({response.status_code}): {response.text}")
        return None

    try:
        payload = response.json()
    except Exception:  # noqa: BLE001
        print("[smoke] Login response was not JSON.")
        return None

    token = payload.get("token") or payload.get("access_token") or payload.get("jwt")
    if not token:
        print("[smoke] No token in login response.")
        return None

    user = payload.get("user") or payload.get("data") or {}
    user_email = user.get("email") if isinstance(user, dict) else user
    print(f"[smoke] Login OK user={user_email} token={str(token)[:8]}...")
    return token


def call_endpoint(path: str, token: str, params: dict | None = None) -> bool:
    url = f"{API_BASE_URL}{path if path.startswith('/') else f'/{path}'}"
    try:
        response = httpx.get(
            url,
            headers={"Authorization": f"Bearer {token}"},
            params=params or None,
            timeout=TIMEOUT,
        )
    except Exception as exc:  # noqa: BLE001
        print(f"[smoke] GET {url} failed: {exc}")
        return False

    try:
        body = response.json()
    except Exception:  # noqa: BLE001
        body = response.text

    if response.status_code != 200:
        print(f"[smoke] GET {url} -> {response.status_code}")
        print(body)
        return False

    print(f"[smoke] GET {url} -> 200 OK")
    return True


def main() -> int:
    token = login()
    if not token:
        return 1

    ok_summary = call_endpoint("/visits/summary", token)
    ok_latest = call_endpoint("/visits/latest", token, params={"pageSize": 5})

    return 0 if ok_summary and ok_latest else 1


if __name__ == "__main__":
    sys.exit(main())

File: CRM/frontend/scripts/smoke-dashboard.mjs
#!/usr/bin/env node
/**
 * Lightweight smoke test to prove the dashboard endpoints succeed with auth.
 * Requires the FastAPI backend to be running locally.
 */

const apiBase = (process.env.CRM_API_BASE_URL || process.env.VITE_API_BASE_URL || 'http://127.0.0.1:8000/api/v1').replace(/\/$/, '');
const email = process.env.SMOKE_LOGIN_EMAIL || 'admin@example.com';
const password = process.env.SMOKE_LOGIN_PASSWORD || 'password';

const buildUrl = path => {
  if (/^https?:\/\//i.test(path)) return path;
  const normalizedPath = path.startsWith('/') ? path : `/${path}`;
  return `${apiBase}${normalizedPath}`;
};

async function login() {
  const response = await fetch(buildUrl('/auth/login'), {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email, password }),
  });
  const payload = await response.json().catch(() => ({}));
  if (!response.ok) {
    throw new Error(`Login failed (${response.status}): ${payload?.detail || response.statusText}`);
  }
  const token = payload.token || payload.access_token || payload.jwt;
  if (!token) {
    throw new Error('Login response missing token field');
  }
  return { token, user: payload.user || payload.data || payload.payload || payload };
}

async function fetchWithAuth(path, token) {
  const response = await fetch(buildUrl(path), {
    headers: { Authorization: `Bearer ${token}` },
  });
  const payload = await response.json().catch(() => ({}));
  if (!response.ok) {
    throw new Error(`${path} failed (${response.status} ${response.statusText}): ${JSON.stringify(payload)}`);
  }
  return payload;
}

async function main() {
  console.log(`Using API base: ${apiBase}`);
  const { token, user } = await login();
  console.log(`Authenticated as ${user?.email || 'unknown user'}`);

  await fetchWithAuth('/visits/summary', token);
  console.log('✅ /visits/summary responded without error');

  await fetchWithAuth('/visits/latest?pageSize=5', token);
  console.log('✅ /visits/latest?pageSize=5 responded without error');
}

main().catch(error => {
  console.error('SMOKE FAILED:', error.message);
  process.exit(1);
});
